sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3191,8 +3198,10 @@ pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
   case PI_DEVICE_INFO_MAX_MEM_BANDWIDTH:
     // currently not supported in level zero runtime
     return PI_ERROR_INVALID_VALUE;
-  case PI_EXT_ONEAPI_DEVICE_INFO_BFLOAT16:
-    return PI_ERROR_INVALID_VALUE;
+  case PI_EXT_ONEAPI_DEVICE_INFO_BFLOAT16: {
+    // bfloat16 math functions are not yet supported on Intel GPUs.
+    return ReturnValue(bool{false});
+  }
```

Comment:
Is this PI_EXT_ONEAPI_DEVICE_INFO_BFLOAT16 specifically asking for "math functions"? How is this different from "bfloat16" extension being supported just above? If this "math" is something more and distinct then why not to add it to the query name?

Reply:
bfloat16 conversions are now supported everywhere, including host code, and device code. We now treat "bfloat16 support" as support for math functions such as fmax, fmin, fabs. We could change the name of this enum value to PI_EXT_ONEAPI_DEVICE_INFO_BFLOAT16_MATH. It will need corresponding changes in the SYCL runtime and all the plugins.

Reply:
Please do the renaming since you are changing the semantics of this query?
Or, better yet, let the existing query mean "conversions" and the new one mean "math"?

Reply:
We discussed changing the name and decided not to.
If we do, it has a cascading effect (15 more files) and is an ABI breakage because the name ext_oneapi_bfloat16 appears in the symbol dump files. 
We instead decided to use the bfloat16 aspect to mean the math functions alone. The aspect for bfloat16 conversions is no longer needed because those are supported everywhere. The documentation reflects this naming. This has been an experimental feature so far, and as the first supported bfloat16 aspect, this is not a change from something that existed before.

Reply:
I've made a change to leave the bfloat16 device query interface as it was and added a new interface for bfloat16_math_functions. So there is an addition to the Linux symbols but no symbol has been dropped. See if this is satisfactory.

Reply:
Yes, that looks much better to me. Thanks for doing the change.